### DISCUSSION TITLE
connections.connect创建连接时，其中的alias和using参数到底有什么区别？list_aliases为None？

### DISCUSSION THREAD

**Discussion by zyy20191** [milvus-io/milvus#18087] [Q&A and General discussion] [answered]
没太理解两个参数的区别，而且当我用下面两行代码创建alias时，list出来的依然是None，是哪里没有设置正确吗，还是说创建别名只能在新建collection时使用？（我是在创建了collection后想添加alias）
```
utility.create_alias("nlp9",  "test")
utility.list_aliases("nlp9")
```
#更新
我在重启服务后，utility.list_collections()命令打印出了之前create_alias中的别名，我有点搞不清这几个函数的关系了

问题一：alias和using参数的使用场景
问题二：create_alias只能在新建collection时使用吗
问题三：create_alias后，list_aliases为什么仍为None（我如果运行两次create_alias会提示我alias已存在，说明创建成功了，但list没有显示）
问题四：创建多个alias后，在connect后使用using参数指定alias吗？指定不同的alias，是否代表创建了新连接？

**Comment by yhmo** [comment, marked as answer]
connections.connect中的alias参数是指rpc连接的别名，collection构造函数中的using参数指的是内部rpc远程调用时所使用的连接，和collection的alias没有任何关系。

collection的alias是通过create_alias()/alter_alias()/drop_alias()/list_alias()这几个接口来管理。当你赋予一个alias给一个collection之后，你就可以使用这个alias来获得collection_name相同的用途。比如调用utility.has_collection(xxx)，这里的xxx就可以用alias来代替collection_name。

理论上：一个alias只能指向一个collection，但一个collection有可能有多个alias指向它
就这么想，一个人可能有多个外号，但人只有一个。每个外号只能用在一个人身上，不能一个外号用在多人，不然你叫这个外号时就不知道是喊谁。而一个人有多个外号也很正常。

list_aliases为什么仍为None，可能是你用的版本是23天以前的，最近有个改动影响list_alias(): https://github.com/milvus-io/milvus/pull/17516，这个改动之后的版本应该是好的。

PS：以后尽量用英文提问，因为用英文我只用答一遍。用中文其他国家的看不懂，到时他们可能又会用英文问一遍，我又得答一次。
